runtime/src/main/java/org/corfudb/runtime/view/BatchSequencerView.java

```diff
@@ -0,0 +1,252 @@
+package org.corfudb.runtime.view;
+
+import com.google.common.collect.MultimapBuilder;
+import com.google.common.collect.SetMultimap;
+import com.google.common.collect.Sets;
+import com.google.common.collect.Table;
+import com.google.common.hash.HashCode;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.Data;
+import org.corfudb.protocols.wireprotocol.BatchTokenRequest;
+import org.corfudb.protocols.wireprotocol.BatchTokenResponse;
+import org.corfudb.protocols.wireprotocol.BatchTokenResponse.BackpointerToken;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.protocols.wireprotocol.TokenType;
+import org.corfudb.protocols.wireprotocol.TxResolutionInfo;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+
+public class BatchSequencerView extends SequencerView {
+
+    @Data
+    private static class BatchSequencerOperation {
+
+        final boolean next;
```

Comment:
This will restrict the current behavior, where a write can request multiple tokens. Why not just use tokenNum and infer "next".

Reply:
No one acquires more than one token, and there doesn't seem to be a use case for it.